experimental/dds/tree2/src/feature-libraries/editable-tree-2/proxies/proxies.ts

```diff
@@ -225,11 +225,22 @@ const asContextuallyTypedData = (value: ProxyNodeUnion<AllowedTypes, "javaScript
 // typed data prior to forwarding to 'LazySequence.insert*()'.
 function itemsAsContextuallyTyped(
 	iterable: Iterable<ProxyNodeUnion<AllowedTypes, "javaScript">>,
-): Iterable<ContextuallyTypedNodeData> {
-	// If the iterable is not already an array, copy it into an array to use '.map()' below.
-	const asArray = Array.isArray(iterable) ? iterable : Array.from(iterable);
+): [items: Iterable<ContextuallyTypedNodeData>, count: number] {
+	const asArray = Array.isArray(iterable)
+		? iterable.map(asContextuallyTypedData)
+		: Array.from(iterable, asContextuallyTypedData);
 
-	return asArray.map(asContextuallyTypedData);
+	return [asArray, asArray.length];
```

Comment:
Nit: Last time I benchmarked, '{ array: asArray, length: asArray.length }' was faster (probably node 12-14).

This is because V8 wasn't smart enough to elide type and/or bounds checking when extracting result[0] and result[1], even though result was obviously length=2.